### DISCUSSION TITLE
Can the data be passed as a parameter through the __call__ function of type Objective?

### DISCUSSION THREAD

**Discussion by daihaozxn** [optuna/optuna#4676] [Q&A] [answered]
Can the data be passed as a parameter through the __call__ function of type Objective?
For example:
```
import optuna

class Objective:
    def __init__(self, parameters):
	self.parameters = parameters
    def __call__(self, trial, data):
	#processing code
	return XXX

study = optuna.create_study()
study.optimize(func=Objective(parameters=parameters)(data=data), n_trials=10)
```
Executing this code above will throw an exception: __call__() missing 1 required positional argument: 'trial'
How should I modify it, or where can I find examples for reference

**Comment by nzw0301** [comment, marked as answer]
Yes, see https://optuna.readthedocs.io/en/stable/faq.html#how-to-define-objective-functions-that-have-own-arguments.

**Reply by daihaozxn** [reply]
> Yes, see https://optuna.readthedocs.io/en/stable/faq.html#how-to-define-objective-functions-that-have-own-arguments.

Thank you for your prompt answer.Does it have to pass data through the __init__ function of the Objective class?
cannot pass through the __call__ function

**Reply by nzw0301** [reply]
No, the second example in the FAQ section I shared is what you want.

**Reply by daihaozxn** [reply]
> No, the second example in the FAQ section I shared is what you want.
OK. I got it. Thank you very much.
